pulseapi/profiles/models.py

```diff
@@ -24,6 +90,95 @@ class UserProfile(models.Model):
         through='profiles.UserBookmarks'
     )
 
+    # The name field is an alternative name to be used if the
+    # associated user(s) don't want to expose their login-indicated
+    # name instead.
+    #
+    # Examples of this are nicknames, pseudonyms, and org names
+    custom_name = models.CharField(
+        max_length=500,
+        blank=True
+    )
+
+    custom_name.short_description = 'Custom user name'
+
+    # Accessing the Profile-indicated name needs various stages
+    # of fallback.
+    def name(self):
+        # blank values, including pure whitespace, don't count:
+        if not self.custom_name:
+            return self.user.name
+        if not self.custom_name.strip():
+            return self.user.name
+
+        # anything else does count.
+        return self.custom_name
+
+    name.short_description = 'Name that will show'
+
+    # This flag marks whether or not this profile applies to
+    # "A human being", or a group of people (be that a club, org,
+    # institution, etc. etc.)
+    is_group = models.BooleanField(
+        default=False
+    )
+
+    is_group.short_description = 'This is a group profile.'
+
+    # Thumbnail image for this user; their "avatar" even though we
+    # do not have anything on the site right now where avatars
+    # come into play.
+    thumbnail = models.ImageField(
+        max_length=2048,
+        upload_to=entry_thumbnail_path,
+        blank=True
+    )
+
+    def thumbnail_image_tag(self):
+        if not self.thumbnail:
+            return format_html('<span>No image to preview</span>')
+
+        media_url = settings.MEDIA_URL
+
+        if settings.USE_S3:
+            media_url = 'https://{domain}/{bucket}/'.format(
+                domain=settings.AWS_S3_CUSTOM_DOMAIN,
+                bucket=settings.AWS_LOCATION
+            )
+
+        html = '<img src="{media_url}{src}" style="width:25%">'.format(
+            media_url=media_url,
+            src=self.thumbnail
+        )
+
+        return format_html(html)
+
+    thumbnail_image_tag.short_description = 'Thumbnail preview'
+
+    # Which issues does this user care about/are they involved in?
+    issues = models.ManyToManyField('issues.Issue')
+
+    # we allow users to indicate several possible predefined service URLs
+    twitter = models.URLField(
+        max_length=2048,
+        blank=True
```

Comment:
hm, makemigrations did not complain about that, doesn't `blank=true` set up a default value? (null is not allowed, but blank is, so when it creates one it'll just be whatever is an empty url field, presumably a blank string?)